shared/chat/delete-history-warning/index.js

```diff
@@ -1,55 +1,89 @@
 // @flow
 import React from 'react'
-import {Box, Button, HeaderOnMobile, Icon, Text} from '../../common-adapters'
-import {globalColors, globalMargins, globalStyles, isMobile} from '../../styles'
+import {Box, Button, HeaderOnMobile, Icon, MaybePopup, Text} from '../../common-adapters'
+import {globalColors, globalMargins, globalStyles, isMobile, platformStyles} from '../../styles'
 
 type Props = {
   errorText: string,
   name: string,
   onBack: () => void,
+  onCancel: () => void,
   onDeleteHistory: () => void,
-  timestamp: string,
 }
 
-const DeleteHistoryWarning = ({errorText, name, onBack, timestamp, onDeleteHistory}: Props) => (
-  <Box
-    style={{
-      ...globalStyles.flexBoxColumn,
-      ...stylePadding,
-      alignItems: 'center',
-      justifyContent: 'center',
-      backgroundColor: globalColors.white,
-      padding: globalMargins.small,
-    }}
-  >
-    <Icon type={isMobile ? 'icon-message-deletion-64' : 'icon-message-deletion-48'} />
-    <Text style={{padding: globalMargins.small}} type="Header">
-      Delete this message + everything above?
-    </Text>
-    <Text style={{padding: globalMargins.small}} type="Body">
-      You are about to delete <Text type="BodySemibold">all messages up to {timestamp}</Text>. For everyone.
-    </Text>
-    <Box style={{...globalStyles.flexBoxRow, marginTop: globalMargins.xlarge}}>
-      <Button type="Secondary" style={{marginLeft: globalMargins.tiny}} onClick={onBack} label="Cancel" />
-      <Button
-        type="Danger"
-        style={{marginLeft: globalMargins.tiny}}
-        onClick={onDeleteHistory}
-        label="Yes, delete these messages"
-      />
+const cancelButton = (onClick: () => void) => {
+  return <Button type="Secondary" style={styleButton} onClick={onClick} label="Cancel" fullWidth={isMobile} />
+}
+
+const DeleteHistoryWarning = ({errorText, name, onCancel, onBack, onDeleteHistory}: Props) => (
+  <MaybePopup onClose={onCancel || onBack}>
+    <Box
+      style={{
+        ...globalStyles.flexBoxColumn,
+        ...stylePadding,
+        alignItems: 'center',
+        justifyContent: 'center',
+        backgroundColor: globalColors.white,
+        padding: globalMargins.small,
+        maxWidth: 560,
+      }}
+    >
+      <Icon type={isMobile ? 'icon-message-deletion-64' : 'icon-message-deletion-48'} />
+      <Text style={{padding: globalMargins.small}} type="Header">
+        Delete conversation history?
+      </Text>
+      <Text style={{padding: globalMargins.small}} type="Body">
+        You are about to delete all the messages in this conversation. For everyone.
+      </Text>
+      <Box style={styleButtonBox}>
+        {!isMobile && cancelButton(onCancel || onBack)}
```

Comment:
You can just style this with flex direction instead of putting in the top or bottom. Column-reverse etc